src/psyclone/tests/psyGen_test.py

```diff
@@ -2621,8 +2631,465 @@ def test_binaryoperation_can_be_printed():
+        real(wp), dimension(:,:) :: f3
+        f2 = f1 + 1
+      end subroutine dummy_code
+    end module dummy_mod
+    '''
+    ast1 = fpapi.parse(DUMMY_KERNEL_METADATA, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test exception for unmatching argument list
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert ("Unexpected kernel AST. The argument list of 'dummy_code'"
+            " do not match the variable declarations.") in str(error.value)
+
+
+def test_fparser2astprocessor_process_declarations():
+    '''Test that process_declarations method of fparse2astprocessor
+    converts the fparser2 decalations to symbols in the provided
+    parent Kernel Schedule.
+    '''
+    from fparser.two.parser import ParserFactory
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Specification_Part
+    f2008parser = ParserFactory().create(std="f2008")
+    fake_parent = KernelSchedule("dummy_schedule")
+    processor = Fparser2ASTProcessor()
+
+    # Test simple declarations
+    reader = FortranStringReader("integer :: l1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("l1").name == 'l1'
+    assert fake_parent.symbol_table.lookup("l1").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("l1").shape == []
+    assert fake_parent.symbol_table.lookup("l1").access == 'local'
+
+    reader = FortranStringReader("Real      ::      l2")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("l2").name == "l2"
+    assert fake_parent.symbol_table.lookup("l2").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("l2").shape == []
+    assert fake_parent.symbol_table.lookup("l2").access == 'local'
+
+    # RHS array specifications are not supported
+    reader = FortranStringReader("integer :: l1(4)")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    with pytest.raises(InternalError) as error:
+        processor.process_declarations(fake_parent, [fparser2specification])
+    assert ("Array specifications after the variable name are not "
+            "supported.") in str(error.value)
+
+    # Initialisations are not supported
+    reader = FortranStringReader("integer :: l1 = 1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    with pytest.raises(InternalError) as error:
+        processor.process_declarations(fake_parent, [fparser2specification])
+    assert ("Array initializations on the declaration statements are not "
```

Comment:
Strictly speaking this is not an array initialisation. Please could you correct the error message?

Reply:
Done